gcloud/datastore/helpers.py

```diff
@@ -73,6 +73,47 @@ def find_true_dataset_id(dataset_id, connection):
     return returned_pb.key.partition_id.dataset_id
 
 
+def _get_meaning(value_pb, is_list=False):
+    """Get the meaning from a protobuf value.
+
+    :type value_pb: :class:`gcloud.datastore._entity_pb2.Value`
+    :param value_pb: The protobuf value to be checked for an
+                     associated meaning.
+
+    :type is_list: bool
+    :param is_list: Boolean indicating if the ``value_pb`` contains
+                    a list value.
+
+    :rtype: int
+    :returns: The meaning for the ``value_pb`` if one is set, else
+              :data:`None`.
+    :raises: :class:`ValueError <exceptions.ValueError>` if a list value
+             has disagreeing meanings (in sub-elements) or has some
+             elements with meanings and some without.
+    """
+    meaning = None
+    if is_list:
+        all_meaning = True
```

Comment:
Here's how I **KNOW** it's contrived. I wrote it 25 days ago and have to think hard to tell what it's doing. Digging in now.